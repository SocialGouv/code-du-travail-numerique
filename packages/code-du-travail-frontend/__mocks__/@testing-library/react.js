/* eslint-disable import/export */
import { theme } from "@socialgouv/cdtn-ui";
import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
