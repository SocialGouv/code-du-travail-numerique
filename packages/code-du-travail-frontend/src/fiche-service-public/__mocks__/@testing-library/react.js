import React from "react";
import PropTypes from "prop-types";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "@socialgouv/cdtn-ui";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
