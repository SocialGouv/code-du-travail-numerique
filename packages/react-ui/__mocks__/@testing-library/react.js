import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

import { colors } from "../../src/theme";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
