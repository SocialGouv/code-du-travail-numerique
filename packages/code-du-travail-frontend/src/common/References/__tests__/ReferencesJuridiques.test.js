import { render } from "@testing-library/react";
import React from "react";

import ReferencesJuridiques from "../index";

const references = [
  {
    slug: "l1244-3",
    title: "L1244-3",
    type: "code_du_travail",
  },
  {
    slug: "r1244-4",
    title: "R1244-4",
    type: "code_du_travail",
  },
  {
    title: "Article L3121-44 du JO",
    type: "external",
    url: "https://article.jo/L3121-44",
  },
  // the two below are not linked to any bloc
  {
    title: "Article foo du JPO",
    type: "external",
    url: "https://article.jo/foo",
  },
  {
    title: "Convention foo",
    type: "external",
    url: "https://ma-convention-collective/ccc-foo",
  },
];

describe("<ReferencesJuridiques />", () => {
  it("should render", () => {
    const { container } = render(
      <ReferencesJuridiques references={references} />
    );
    expect(container).toMatchSnapshot();
  });
});
