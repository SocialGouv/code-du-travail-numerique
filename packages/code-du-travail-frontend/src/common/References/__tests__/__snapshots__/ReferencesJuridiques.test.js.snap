// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReferencesJuridiques /> should render 1`] = `
.c0 {
  padding: 2rem 2.4rem;
  color: #3e486e;
  border-radius: 0.6rem;
  border: 1px solid #bbcadf;
  background-color: #fff;
}

.c0 > *:first-child {
  margin-top: 0;
}

.c0 > *:last-child {
  margin-bottom: 0;
}

.c4 {
  position: relative;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  color: #7994d4;
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
  -webkit-transition: -webkit-transform 250ms linear;
  -webkit-transition: transform 250ms linear;
  transition: transform 250ms linear;
}

[aria-expanded="true"] .c3,
[aria-selected="true"] .c4 {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c15 {
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  width: auto;
}

.c14 {
  -webkit-flex: 0 0 2.8rem;
  -ms-flex: 0 0 2.8rem;
  flex: 0 0 2.8rem;
  width: 2.8rem;
  height: 2.1rem;
  margin: 0 1.6rem 0 0;
  padding-top: 0.6rem;
  color: #f66663;
  -webkit-transition: -webkit-transform 250ms linear;
  -webkit-transition: transform 250ms linear;
  transition: transform 250ms linear;
}

.c10:hover .c13 {
  -webkit-transform: translateX(4px);
  -ms-transform: translateX(4px);
  transform: translateX(4px);
}

.c6 {
  color: #3e486e;
  line-height: 1.25;
  font-size: 1.8rem;
  font-weight: 600;
}

.c7 {
  max-height: 0px;
  opacity: 0;
  overflow: auto;
  padding: 0;
  -webkit-transition: opacity 0.5s ease-in-out;
  transition: opacity 0.5s ease-in-out;
}

.c2 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  overflow: hidden;
}

.c2:hover,
.c2:focus,
.c2:focus-within,
.c2[aria-expanded="true"] {
  color: #3e486e;
}

.c5 {
  margin-left: 1rem;
}

.c8 {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.c12 {
  padding: 0.8rem 0;
  font-weight: normal;
}

.c9 {
  padding-left: 0;
}

.c1 {
  padding-top: 0;
  padding-bottom: 0;
}

@media (max-width:600px) {
  .c0 {
    padding: 1.6rem 1rem;
  }
}

@media print {
  .c0 {
    padding: 0 5pt;
    border: none;
  }
}

@media (max-width:600px) {
  .c14 {
    -webkit-flex: 0 0 2rem;
    -ms-flex: 0 0 2rem;
    flex: 0 0 2rem;
    width: 2rem;
    height: 1.7rem;
    margin: 0 1rem 0 0;
  }
}

@media (max-width:600px) {
  .c1 {
    padding-top: 0;
    padding-bottom: 0;
  }
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      <div
        aria-expanded="false"
        class="c2"
        role="button"
      >
        <svg
          aria-hidden="true"
          class="c3 c4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
        <div
          class="c5"
        >
          <span
            class="c6"
            font-size="hsmall"
            font-weight="600"
          >
            Références juridiques concernées :
          </span>
        </div>
      </div>
      <div
        class="c7"
      >
        <ul
          class="c8 c9"
        >
          <li>
            <a
              class="c10 c11 c12"
              href="/code-du-travail/l1244-3"
            >
              <svg
                aria-hidden="true"
                class="c13 c14"
                fill="none"
                viewBox="0 0 28 15"
              >
                <path
                  d="M27.875 7.153l-6-6.284a.526.526 0 00-.711-.023.496.496 0 00-.024.688l5.204 5.468H.5c-.276 0-.5.217-.5.484s.224.483.5.483h25.844l-5.204 5.469a.493.493 0 00.024.687.52.52 0 00.711-.023l6-6.284a.506.506 0 000-.665z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="c15"
              >
                Article L1244-3 du Code du travail
              </span>
            </a>
          </li>
          <li>
            <a
              class="c10 c11 c12"
              href="/code-du-travail/r1244-4"
            >
              <svg
                aria-hidden="true"
                class="c13 c14"
                fill="none"
                viewBox="0 0 28 15"
              >
                <path
                  d="M27.875 7.153l-6-6.284a.526.526 0 00-.711-.023.496.496 0 00-.024.688l5.204 5.468H.5c-.276 0-.5.217-.5.484s.224.483.5.483h25.844l-5.204 5.469a.493.493 0 00.024.687.52.52 0 00.711-.023l6-6.284a.506.506 0 000-.665z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="c15"
              >
                Article R1244-4 du Code du travail
              </span>
            </a>
          </li>
          <li>
            <a
              aria-label="Article L3121-44 du JO (Nouvelle fenêtre)"
              class="c10 c11 c12 no-after"
              href="https://article.jo/L3121-44"
              rel="noopener noreferrer nofollow"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="c13 c14"
                fill="none"
                viewBox="0 0 28 15"
              >
                <path
                  d="M27.875 7.153l-6-6.284a.526.526 0 00-.711-.023.496.496 0 00-.024.688l5.204 5.468H.5c-.276 0-.5.217-.5.484s.224.483.5.483h25.844l-5.204 5.469a.493.493 0 00.024.687.52.52 0 00.711-.023l6-6.284a.506.506 0 000-.665z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="c15"
              >
                Article L3121-44 du JO
              </span>
            </a>
          </li>
          <li>
            <a
              aria-label="Article foo du JPO (Nouvelle fenêtre)"
              class="c10 c11 c12 no-after"
              href="https://article.jo/foo"
              rel="noopener noreferrer nofollow"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="c13 c14"
                fill="none"
                viewBox="0 0 28 15"
              >
                <path
                  d="M27.875 7.153l-6-6.284a.526.526 0 00-.711-.023.496.496 0 00-.024.688l5.204 5.468H.5c-.276 0-.5.217-.5.484s.224.483.5.483h25.844l-5.204 5.469a.493.493 0 00.024.687.52.52 0 00.711-.023l6-6.284a.506.506 0 000-.665z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="c15"
              >
                Article foo du JPO
              </span>
            </a>
          </li>
          <li>
            <a
              aria-label="Convention foo (Nouvelle fenêtre)"
              class="c10 c11 c12 no-after"
              href="https://ma-convention-collective/ccc-foo"
              rel="noopener noreferrer nofollow"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="c13 c14"
                fill="none"
                viewBox="0 0 28 15"
              >
                <path
                  d="M27.875 7.153l-6-6.284a.526.526 0 00-.711-.023.496.496 0 00-.024.688l5.204 5.468H.5c-.276 0-.5.217-.5.484s.224.483.5.483h25.844l-5.204 5.469a.493.493 0 00.024.687.52.52 0 00.711-.023l6-6.284a.506.506 0 000-.665z"
                  fill="currentColor"
                />
              </svg>
              <span
                class="c15"
              >
                Convention foo
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
